sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessMessageEventArgs.cs

```diff
@@ -24,6 +25,12 @@ public class ProcessMessageEventArgs : EventArgs
         /// </summary>
         public CancellationToken CancellationToken { get; }
 
+        /// <summary>
+        /// Indicates whether the user has attempted to settle the message as part of their callback.
+        /// If they have done so, we will not autocomplete.
+        /// </summary>
+        internal bool UserSettled { get; set; }
```

Comment:
Did you check if somehow we can access [Settled](https://github.com/Azure/azure-amqp/blob/master/src/Delivery.cs#L45) property after settling a message, then we don't need UserSettled property.

I've checked but couldn't find a way to access Settled property when we settle the messages :disappointed: